### PR TITLE
Make gnome-twitch ready for gobject-introspection

### PR DESCRIPTION
**Pull request by grindhold**
I did a bunch of things in order to make it possible to write player-backends in other languages than C and make full-fledged use of GObject Introspection.

### CODE REVIEW

File: src/meson.build [hunk 1]
```diff
@@ -127,12 +129,22 @@ gt_library_dep = declare_dependency(
...
   libraries : gt_library,
-  name : 'gnome-twitch',
+  name : 'gnome-twitch-' + api,
   description : 'Library used for building GNOME Twitch plugins',
   requires : ['gtk+-3.0', 'libpeas-1.0'],
   install : true,
   version : meson.project_version())
 
+gnome.generate_gir(gt_library,
+            sources: [meson.current_source_dir()+'/../include/gnome-twitch/gt-player-backend.h'],
```
Comment (TingPing) [inline src/meson.build]: Should be using 2 spaces for indentation.

Comment (TingPing) [inline src/meson.build]: You should never be using `meson.current_source_dir()` you should be able to use a relative path or worst case a `files()` call.

**Comment by grindhold** [comment]
this should now be ready to rumble

File: src/meson.build [hunk 2]
```diff
@@ -136,13 +136,14 @@ pkg.generate(
...
-            namespace: 'Gt',
-            dependencies: deps_gt_library,
-            install: true,
-            install_dir_gir: get_option('datadir') + '/gir-1.0',
-            install_dir_typelib: get_option('libdir') + '/girepository-1.0'
+  sources: ['../include/gnome-twitch/gt-player-backend.h'],
+  nsversion: api,
+  namespace: 'Gt',
+  dependencies: deps_gt_library,
+  extra_args: ['--include','Gtk-3.0'],
```
Comment (TingPing) [inline src/meson.build]: `includes: ['Gtk-3.0']`

**Comment by grindhold** [comment]
fixed

File: src/meson.build [hunk 3]
```diff
@@ -136,14 +136,14 @@ pkg.generate(
...
+  sources: [join_paths('..', 'include', 'gnome-twitch', 'gt-player-backend.h')],
   nsversion: api,
   namespace: 'Gt',
   dependencies: deps_gt_library,
   includes: ['Gtk-3.0'],
   install: true,
-  install_dir_gir: get_option('datadir') + '/gir-1.0',
-  install_dir_typelib: get_option('libdir') + '/girepository-1.0'
+  install_dir_gir: join_paths(get_option('datadir'), 'gir-1.0'),
+  install_dir_typelib: join_paths(get_option('libdir'), '/girepository-1.0')
```
Comment (TingPing) [inline src/meson.build]: This is now broken, drop the leading `/`.

Comment (grindhold) [inline src/meson.build]: done. thx for pointing out.

**Comment by vinszent** [comment]
Looks good! Thanks for writing this and thanks @TingPing for reviewing.

**Pull request merged**